lib/std/process/Child.zig

```diff
@@ -1270,61 +1219,148 @@ fn windowsCreateProcessPathExt(
+    child: *Child,
+    app_name: [:0]const u16,
+    cmd_line: []const u16,
+    envp: ?[]const u16,
+    cwd: ?[]const u16,
+    handles: StdHandles,
 ) !void {
-    // TODO the docs for environment pointer say:
-    // > A pointer to the environment block for the new process. If this parameter
-    // > is NULL, the new process uses the environment of the calling process.
-    // > ...
-    // > An environment block can contain either Unicode or ANSI characters. If
-    // > the environment block pointed to by lpEnvironment contains Unicode
-    // > characters, be sure that dwCreationFlags includes CREATE_UNICODE_ENVIRONMENT.
-    // > If this parameter is NULL and the environment block of the parent process
-    // > contains Unicode characters, you must also ensure that dwCreationFlags
-    // > includes CREATE_UNICODE_ENVIRONMENT.
-    // This seems to imply that we have to somehow know whether our process parent passed
-    // CREATE_UNICODE_ENVIRONMENT if we want to pass NULL for the environment parameter.
-    // Since we do not know this information that would imply that we must not pass NULL
-    // for the parameter.
-    // However this would imply that programs compiled with -DUNICODE could not pass
-    // environment variables to programs that were not, which seems unlikely.
-    // More investigation is needed.
-    return windows.CreateProcessW(
-        app_name,
-        cmd_line,
+    const parent_params = windows.peb().ProcessParameters;
+    const app_name_us = try windows.UNICODE_STRING.init(@constCast(app_name));
+    const image_name = try windows.wToPrefixedFileW(app_name, .{
+        .dir = if (cwd) |path| .{ .wtf16_path = path } else .cwd,
+        .allow_relative = false,
+    });
+    const env = env: {
+        const min_env = [1]u16{0} ** 2;
+        const env = envp orelse parent_params.Environment[0..@divExact(
+            parent_params.EnvironmentSize,
+            @sizeOf(windows.WCHAR),
+        )];
+        break :env if (env.len >= min_env.len) env else &min_env;
+    };
+    var create_info: windows.PS.CREATE_INFO = .{ .Info = .{ .InitialState = .{
+        .InitFlags = .{
+            .WriteOutputOnExit = true,
+            .DetectManifest = true,
+            .ProhibitedImageCharacteristics = 0x2000,
+        },
+        .AdditionalFileAccess = .{ .SPECIFIC = .{ .FILE = .{
+            .READ_DATA = true,
+            .READ_ATTRIBUTES = true,
+        } } },
+    } } };
+    defer create_info.deinit();
+    var handle_list: [StdHandles.fields.len]windows.HANDLE = undefined;
+    var handle_list_len: usize = 0;
+    inline for (StdHandles.fields) |field| {
+        if (handles.get(field)) |handle| {
+            if (mem.indexOfScalar(windows.HANDLE, handle_list[0..handle_list_len], handle) == null) {
+                handle_list[handle_list_len] = handle;
+                handle_list_len += 1;
+            }
+        }
+    }
+    switch (windows.ntdll.NtCreateUserProcess(
```

Comment:
Use `NtCreateUserProcess` at your own peril. Relevant quote from @zodiacon's "Windows Native API Programming":

> There have been several attempts in the Infosec community to utilize `RtlCreateUserProcess` and/or `NtCreateUserProcess` to allow running Windows subsystem applications, with varying degrees of success. Part of the problem is the need to communicate with the Windows Subsystem process (`csrss.exe`) to notify it of the new process and thread. This turns out to be fragile, as different Windows versions may have somewhat different expectations.

Essentially: you're one the hook if this breaks on newer Windows versions and/or Wine updates.